### PR TITLE
Support basic sriov-fec test

### PR DESCRIPTION
**Pull request by SchSeba**
* Bump intel operator
* add a basic sriov-fec test to check we are able to configure the acc100 card and the resources are exposed in the cluster.

waiting for  https://github.com/openshift-kni/cnf-features-deploy/pull/444

### CODE REVIEW

File: cnf-tests/testsuites/pkg/utils/consts.go [hunk 1]
```diff
@@ -114,4 +116,10 @@ const (
 	SriovFecNodeConfigCRDName = "sriovfecnodeconfigs.sriovfec.intel.com"
 	// SriovFecClusterConfigCRDName contains the name of the SriovFecCluster config policies CRD
 	SriovFecClusterConfigCRDName = "sriovfecclusterconfigs.sriovfec.intel.com"
+	// SriovFecClisterConfigName contains the name of the SriovFecCluster config policies allow by the operator
+	// https://github.com/open-ness/openshift-operator/blob/main/sriov-fec/controllers/sriovfecclusterconfig_controller.go#L87
+	SriovFecClisterConfigName = "config"
```
Comment (oribon) [inline cnf-tests/testsuites/pkg/utils/consts.go:121]: nit: typo (Clister)

File: cnf-tests/testsuites/e2esuite/sriov-fec/sriov-fec.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/utils"
+)
+
+var (
+	nodeName string
+	numVfs   int
```
Comment (oribon) [inline cnf-tests/testsuites/e2esuite/sriov-fec/sriov-fec.go:26]: does this get a value somewhere or it's supposed to stay 0?

**Comment by SchSeba** [comment]
@oribon can you please make another review?

File: cnf-tests/testsuites/e2esuite/fec/fec.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+	if err != nil {
+		return err
+	}
+
+	return client.Client.Delete(context.TODO(), sriovFecCluster)
+}
+
+func createSriovFecClusterObject() error {
+	nodesWithAcc100, err := getSriovFecNodes()
```
Comment (fedepaol) [inline cnf-tests/testsuites/e2esuite/fec/fec.go:76]: Can you split this in "findNodeWithFec" or something?
Here we set the global as a side effect, and it's not clear where that is coming from.
I'd prefer the global to be set in beforeEach and to use it as a parameter for this

File: cnf-tests/testsuites/e2esuite/fec/fec.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	sriovfecv1 "github.com/open-ness/openshift-operator/sriov-fec/api/v1"
+
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/utils"
+)
+
+var (
+	nodeName string
```
Comment (fedepaol) [inline cnf-tests/testsuites/e2esuite/fec/fec.go:25]: Can this be local to the test?

File: cnf-tests/testsuites/e2esuite/fec/fec.go [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+		err := cleanSriovFecClusterObject()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = createSriovFecClusterObject()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		err := cleanSriovFecClusterObject()
```
Comment (fedepaol) [inline cnf-tests/testsuites/e2esuite/fec/fec.go:44]: Do we need to clean before and after?

Comment (fedepaol) [reply cnf-tests/testsuites/e2esuite/fec/fec.go:44]: Also, would it make sense to add these new resources/namespaces to the reporter?

Comment (fedepaol) [reply cnf-tests/testsuites/e2esuite/fec/fec.go:44]: Also also, would it make sense to export a Clean function to be called when the suite shuts down?

File: cnf-tests/testsuites/pkg/utils/consts.go [hunk 6]
```diff
@@ -114,4 +116,10 @@ const (
 	SriovFecNodeConfigCRDName = "sriovfecnodeconfigs.sriovfec.intel.com"
 	// SriovFecClusterConfigCRDName contains the name of the SriovFecCluster config policies CRD
 	SriovFecClusterConfigCRDName = "sriovfecclusterconfigs.sriovfec.intel.com"
+	// SriovFecClusterConfigName contains the name of the SriovFecCluster config policies allow by the operator
+	// https://github.com/open-ness/openshift-operator/blob/main/sriov-fec/controllers/sriovfecclusterconfig_controller.go#L87
+	SriovFecClusterConfigName = "config"
```
Comment (fedepaol) [inline cnf-tests/testsuites/pkg/utils/consts.go:121]: If these are local only to the fec test, let's put them in that go file.
Use them here only if they are shared between packages

File: cnf-tests/testsuites/e2esuite/fec/fec.go [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+	sriovfecv1 "github.com/open-ness/openshift-operator/sriov-fec/api/v1"
+
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/utils"
+)
+
+var _ = FDescribe("fec", func() {
```
Comment (fedepaol) [inline cnf-tests/testsuites/e2esuite/fec/fec.go:24]: remove the F

File: cnf-tests/testsuites/e2esuite/fec/fec.go (hunk 3 above)
Comment (fedepaol) [reply cnf-tests/testsuites/e2esuite/fec/fec.go:97]: I'd prefer (but feel free to leave it as it is) to have this take the node as a parameter, so wrapping all the node selection logic in a single function and calling it in beforeEach

File: cnf-tests/testsuites/pkg/utils/consts.go (hunk 6 above)
Comment (fedepaol) [reply cnf-tests/testsuites/pkg/utils/consts.go:121]: Move the constants :-)

File: cnf-tests/testsuites/e2esuite/fec/fec.go (hunk 3 above)
Comment (SchSeba) [reply cnf-tests/testsuites/e2esuite/fec/fec.go:94]: done I split the function

**Pull request merged**